packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -343,6 +352,36 @@ The following table lists the set of triggers available, and their corresponding
 For more information on the function of these triggers and how to configure them, read [User Pool Workflows with
 Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
 
+You may need to provide your Lambda trigger with permissions to perform operations against your user pool.
+Permission can be provided by through the `attachInlinePolicy` API on the Lambda function.
+
+```ts
+import * as lambda from '@aws-cdk/aws-lambda';
+
+const postAuthFn = new lambda.Function(this, 'postAuthFn', {
+  code: lambda.Code.fromInline('foo'),
+  runtime: lambda.Runtime.NODEJS_12_X,
+  handler: 'index.handler',
+});
+
+const userpool = new UserPol(this, 'myuserpool', {
+  lambdaTriggers: {
+    postAuthentication: postAuthFn,
+  },
+});
+
+// provide permissions to describe the user pool
+fn.role?.attachInlinePolicy(new Policy(stack, 'userpool-policy', {
+  statements: [new PolicyStatement({
+    actions: ['cognito-idp:DescribeUserPool'],
+    resources: [userpool.userPoolArn],
+  })],
+}));
+```
+
+⚠️ Using the `attachToRolePolicy` API to provide permissions to a Lambda trigger will result in a circular dependency error:
+> Circular dependency between resources: [pool056F3F7E, fnPostAuthFnCognitoA630A2B1, ...]
```

Comment:
Would be nice to state this before providing the work around.

Reply:
agreed. moved this up to highlight it before offering the workaround.